src/nomadic/summarize/main.py

```diff
@@ -0,0 +1,805 @@
+    log.info("Input parameters:")
+    log.info(f"  Summary Name: {summary_name}")
+    if not no_master_metadata:
+        log.info(f"  Master metadata: {meta_data_path}")
+    else:
+        log.info("  No master metadata will be used.")
+    log.info(f"  Setting file: {settings_file_path}")
+    log.info(f"  Found {len(expt_dirs)} experiment directories.")
+    for expt_dir in expt_dirs:
+        check_complete_experiment(expt_dir)
+    log.info("  All experiments are complete.")
+
+    settings: Settings = Settings()
+
+    if settings_file_path.exists():
+        settings = load_settings(settings_file_path)
+        log.info(f"  Loaded summary settings from {settings_file_path}.")
+
+    # CHECK METADATA IS VALID
+    # TODO:
+    # - Should I already interrogate geospatial information?
+    # - Where should I compute throughput information?
+    dfs = []
+    for expt_dir in expt_dirs:
+        metadata_csv = get_metadata_csv(expt_dir)
+        try:
+            parser = ExtendedMetadataTableParser(metadata_csv)
+            parser.df.insert(0, "expt_name", os.path.basename(expt_dir))
+        except MetadataFormatError as e:
+            raise MetadataFormatError(
+                f"Metadata format issue in experiment directory {expt_dir}: {e}"
+            ) from e
+        if not dfs:
+            shared_columns = set(parser.df.columns)
+        shared_columns.intersection_update(parser.df.columns)
+        dfs.append(parser.df)
+        # Should I not take all common columns?
+    log.info("  All metadata tables pass completion checks.")
+    log.info(
+        f"  Found {len(shared_columns)} shared columns across all metadata files: {', '.join(shared_columns)}"
+    )
+    fixed_columns = ["expt_name", "barcode", "sample_id", "sample_type"]
+    shared_columns.difference_update(fixed_columns)
+    # for now we use the master metadata file
+    inventory_metadata = pd.concat([df[fixed_columns] for df in dfs])
+    if meta_data_path is not None and not no_master_metadata:
+        master_metadata = pd.read_csv(meta_data_path).rename(
+            columns=get_master_columns_mapping(settings)
+        )
+    else:
+        # create metadata from experiment meta data files
+        shared_columns = ["sample_id"] + list(shared_columns)
+        master_metadata = pd.concat([df[shared_columns] for df in dfs])
+
+    master_metadata = master_metadata.astype(
+        {"sample_id": "str"}
+    )  # ensure sample IDs are strings
+    inventory_metadata = inventory_metadata.astype(
+        {"sample_id": "str"}
+    )  # ensure sample IDs are strings
+    # strip whitespaces from sample IDs
+    inventory_metadata["sample_id"] = inventory_metadata["sample_id"].str.strip()
+    master_metadata["sample_id"] = master_metadata["sample_id"].str.strip()
+
+    # In case sample IDs are numbers, we want to strip leading zeros (this was a problem in Zambia data)
+    inventory_metadata["sample_id"] = inventory_metadata["sample_id"].str.lstrip("0")
```

Comment:
I don't think Nomadic should take on the responsibility of cleaning sample IDs. If the sample IDs have issues with them, we should leave that for users with knowledge of the study to fix.